### ISSUE TITLE
Audio does not connect

### ISSUE THREAD

**Issue by siavash79** [bennyboer/bbb_app#131]
Hi

I did build and install the app on stock (AOSP) android 11

The app never connects audio: I'm not able to hear or speak in the rooms.

Is there any way I can share the logs here?

**Comment by Gleydar** [comment]
Hi, if you want, you can upload the logs in this ticket. That way, everyone working on it can see them. Iirc, they shouldn't contain any identifiable information.

**Comment by siavash79** [comment]
Of course I will. The question is that where exactly are the logs stored?

**Comment by Gleydar** [comment]
Ah, okay, thanks for clarifying. The logs should be in logcat, or, if you build a debug version of the app in Android studio itself

**Comment by siavash79** [comment]
Ok I'm sorry for delay. Apparently I didn't receive github reply notification mail for some reason

here's my logcat output, filtered by flutter.

The strange thing is that logcat shows that sip is connected, but bbb html5 client on my laptop doesn't seem to agree with that (showing audio icon as blank, no audio from phone at all)

One thing to note is that in my server configuration, I've set it to connect audio directly (html5 doesn't ask whether to open listen-only or microphone, and directly proceeds to echo test). I'm not sure if it's related or not

[bbb_appp_flutter_logcat.txt](https://github.com/bennyboer/bbb_app/files/5834180/bbb_appp_flutter_logcat.txt)

**Comment by bennyboer** [comment]
@lukaskirner @Gleydar Can we setup an environment with direct audio connect to test this?

**Comment by siavash79** [comment]
You'll be welcome to test in my environment if you like. My server is up
and running.

You can drop me an email to setup a test room
My address is (siavash at gmail)

**Comment by lukaskirner** [comment]
After testing with @siavash79 system it may be related to #133. In debug build the app crashes with the same error from #133.

**Comment by bennyboer** [comment]
I don't think the problems are related. We should recheck after #133 is fixed.

**Comment by siavash79** [comment]
Sure. I'll keep the room accessible just in case

Uh: While you're there, you may wanna check out what happens if there's no
presentation going on in the meeting (It shows a blank space in the app).
My setup doesn't have a default presentation on meeting start point.

issue:
https://github.com/bennyboer/bbb_app/issues/130

**Comment by bennyboer** [comment]
#133 is fixed, can you recheck? 😄

**Comment by siavash79** [comment]
Unfortunately not

This is what I've done (hopefully this has fetched the fix?)

git fetch origin
flutter build apk --debug
adb install build\app\outputs\flutter-apk\app-debug.apk

**Comment by bennyboer** [comment]
You can check if you have the following in the `pubspec.yaml` file:
```
sip_ua: # ^0.3.4
    git: https://github.com/flutter-webrtc/dart-sip-ua.git # TODO Reset to using version number > 0.3.5 after it is released
```

Then you have the fix 😄

**Comment by siavash79** [comment]
My apologies apparently the source was not updated. I can confirm to have the latest code now

At the time being, the app doesn't even join the meeting anymore (either notifies: you've been removed, or opens the room without loading anything else and doesn't show the participant name in the room on the other side)
[logcat.txt](https://github.com/bennyboer/bbb_app/files/5917968/logcat.txt)

**Comment by bennyboer** [comment]
I will check ASAP (hope I can do it today)

**Comment by andydevmobile** [comment]
I'm facing the same issue with audio connection so please don't mind if I provide some information in this thread.
I've already pull latest commit and add 
`git: https://github.com/flutter-webrtc/dart-sip-ua.git`
in my pubspec.yml, but `CallStateEnum` turn from `STREAM` to `FAILED` with `Code: [408], Cause: Canceled, Reason: SIP ;cause=408 ;text="Request Timeout"`
Beside, I still see myself in the room but with white micro icon, and I can't hear or make any sound
![image](https://user-images.githubusercontent.com/38199906/106745480-bb8f9500-6653-11eb-9845-10fd5339215b.png)
Below is my full log from flutter, hope this help you guys and many thanks for your awesome works.

[log.txt](https://github.com/bennyboer/bbb_app/files/5918309/log.txt)

**Comment by andydevmobile** [comment]
And one more thing, when join in room, it doesn't open an echo test, is it normal? Sorry if it's my mistake.

**Comment by bennyboer** [comment]
Thanks for the information we will try to reproduce it 😄

We skip the echo test as it does not make much sense in our opinion on smartphones.

**Comment by lukaskirner** [comment]
Which BBB Version do you all have?

**Comment by bennyboer** [comment]
After spending a lot of time to find the issue I've found just that the official web app is sending the SIP messages with something like `Via: SIP/2.0/WS 7ghjrpm3fkjl.invalid;branch=z9hG4bK6115426` whereas our BBB server always send `Via: SIP/2.0/WSS 25ma3u024204.invalid;branch=z9hG4bK7259768500000000`.
But the connection on the server is using the wss protocol so I have no Idea why it's behaving like that.

In the log it also states that `Allow self-signed certificate => XXX.XXXX.com:443.`. Maybe that's why the app is acting up

**Comment by bennyboer** [comment]
We have confirmation that the problem occurs on BBB `2.2.31`.

**Comment by bennyboer** [comment]
> 
> 
> After spending a lot of time to find the issue I've found just that the official web app is sending the SIP messages with something like `Via: SIP/2.0/WS 7ghjrpm3fkjl.invalid;branch=z9hG4bK6115426` whereas our BBB server always send `Via: SIP/2.0/WSS 25ma3u024204.invalid;branch=z9hG4bK7259768500000000`.
> But the connection on the server is using the wss protocol so I have no Idea why it's behaving like that.
> 
> In the log it also states that `Allow self-signed certificate => XXX.XXXX.com:443.`. Maybe that's why the app is acting up

Funnily this is exactly the problem.
BBB v2.2.31 expects SIP/2.0/WS despite sending over a WSS connection!
But when I change that it does not work for older BBB Servers anymore 😄 

That is frustrating! I will check whether I can distinguish the versions from the client and either apply to one or the other.

**Comment by siavash79** [comment]
Just a thought: can't you try wss and see if you get a working connection,
and if not, immediately fallback to ws on the same port?

I'm thinking for a way to consider the introduced protocol as unreliable,
regardless of bbb version

**Comment by bennyboer** [comment]
Yes, that's my current proposal I will attempt to use WSS as before and if that fails without the echo test succeeding, it will try again one more time using WS

**Comment by bennyboer** [comment]
I am not really happy with the solution but will commit it anyway to have a solution. We should show notifications for the user to know what the current audio connection status is. Created issue #154 for this.

**Comment by siavash79** [comment]
I can confirm that with #155, the audio does connect on 2.2.31, BUT:

being a single client in the meeting, apparently the app connected 2 times, which made me hear myself (like echo test) in a single-participant room.

**Comment by siavash79** [comment]
> I can confirm that with #155, the audio does connect on 2.2.31, BUT:
> 
> being a single client in the meeting, apparently the app connected 2 times, which made me hear myself (like echo test) in a single-participant room.

Adding to above: since it apparently connected 2 times, the mute button worked on only one of the connections, practically made it impossible to mute

I'll have to add that the first join attempt failed with the app jumping back to login page (apparently sip connected anyway!), and second attempt did actually join the room, opening a second concurrent sip connection.

This can also count as a security thread, having a room seemingly not joined, but sip is transmitting audio without user consent.

**Comment by bennyboer** [comment]
@siavash79 Created new issue #156 for this, will check ASAP!

**Comment by ciftcisamet** [comment]
> I'm facing the same issue with audio connection so please don't mind if I provide some information in this thread.
> I've already pulled latest commit and add
> `git: https://github.com/flutter-webrtc/dart-sip-ua.git`
> in my pubspec.yml, but `CallStateEnum` turn from `STREAM` to `FAILED` with `Code: [408], Cause: Canceled, Reason: SIP ;cause=408 ;text="Request Timeout"`
> Beside, I still see myself in the room but with white micro icon, and I can't hear or make any sound
> ![image](https://user-images.githubusercontent.com/38199906/106745480-bb8f9500-6653-11eb-9845-10fd5339215b.png)
> Below is my full log from flutter, hope this help you guys and many thanks for your awesome works.
> 
> [log.txt](https://github.com/bennyboer/bbb_app/files/5918309/log.txt)

I am facing the same issue. Could you manage to solve this?
